weaver/cli.py

```diff
@@ -328,8 +328,8 @@ class WeaverClient(object):
     monitor_interval = 5    # interval between monitor pooling job status requests
     auth = None  # type: AuthHandler
 
-    def __init__(self, url=None, auth=None):
-        # type: (Optional[str], Optional[AuthHandler]) -> None
+    def __init__(self, url=None, auth=None, session=None):
+        # type: (Optional[str], Optional[AuthHandler], Optional[Session]) -> None
```

Comment:
See PR comment about using other `AuthHandler` alternatives. Seems to me that available ones are sufficient to handle the use case (passing Magpie cookies).

If they are still not sufficient for some reason (please justify if so), I would rather have a `SessionAuthHandler` that reuses the same `auth` parameter instead of adding a new one only for that use case.

Reply:
> Seems to me that available ones are sufficient to handle the use case

Unless I'm missing some other way of using these handlers, I think they cover a different use case (see: https://github.com/crim-ca/weaver/pull/597#issuecomment-1930745774)

> I would rather have a SessionAuthHandler

Actually that would be really easy. We could do something like:

```python
class SessionAuthHandler(AuthBase):
    def __init__(self, session=None):
        self.session = session

    def __call__(self, r):
        r.prepare_cookies(self.session.cookies)
        return r
```